tokens/usecase/token_registry.go

```diff
@@ -0,0 +1,94 @@
+package usecase
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/osmosis-labs/sqs/domain"
+)
+
+// GetTokensFromChainRegistryFunc is a GetTokensFromChainRegistry function signature.
+type GetTokensFromChainRegistryFunc func(chainRegistryAssetsFileURL string) (map[string]domain.Token, string, error)
+
+// GetTokensFromChainRegistry fetches the tokens from the chain registry.
+// It returns a map of tokens by chain denom.
+func GetTokensFromChainRegistry(chainRegistryAssetsFileURL string) (map[string]domain.Token, string, error) {
```

Comment:
Can we issue `response.Body.Close()` only once, and read the `response.Body` once ? Or there is a reason for that? I'm not sure what is the purpose of line 34 - line 39. Could you please explain? Thank you!

Reply:
Oh, that's a good catch, thank you! I have updated response to be read once and then data to be reused for several operations. I am not sure how I have ended up with this code.